### DISCUSSION TITLE
Second NIC not tried for http

### DISCUSSION THREAD

**Discussion by wf2020** [ipxe/ipxe#568] [Q&A] [answered]
I have two network interfaces. Selecting second for booting loads iPXE. After choosing what to start/load over http only net0 is used for connecting, net1 which the server is connected to is only used if link for net0 is down.

Can this be corrected?

**Comment by wf2020** [comment]
No, it does NOT try each NIC. It only tries the first one!
Only if the first one has no link the second one is tried too.

**Reply by NiKiZe** [reply]
Do you have an embedded script or any other modifications? Can you show a screenshot?

**Comment by wf2020** [comment]
I only use this script:

`dhcp`
`chain http://192.168.2.204/pxe/main.ipxe`

At boot the client simply only tries net0 if it is connected and this doesn't succeed (because the server is not there).
If the first NIC is not connected it takes some time till the trial times out, afterwards net1 is used successfully.

**Reply by NiKiZe** [reply]
dhcp is not autoboot and behaves differently.
If you want embedded script then you will have to build the logic. But dhcp || with some Goto's and logic might work.

**Reply by wf2020** [reply]
Okay, thank you so far. I will investigate on that later.

**Reply by wf2020** [reply]
I did another trial.
Interesting enough: if I deactivate the DHCP server on net0 it works!

For me this seems to be a bug. If there is provided an IP address at the wrong(!) subnet and there is another net available this one should be tried too.

**Reply by NiKiZe** [reply]
What exactly is wrong?
https://ipxe.org/cmd/ifconf

If you run dhcp and you get an IP on net0 then it succeeds and is done.
How do you define wrong subnet? And how would iPXE know?
Maybe you are using ipxe.pxe or ipxe.efi, but you are expecting the behaviour that you get with undionly.kpxe or  snponly.efi (only chained interface used)

Why do you think it should behave differently?
Please give enough information so that anyone (with zero knowledge about your setup or configuration) can reproduce.

I asked for a screenshot to try and understand what the actuall issue for you is.

**Reply by NiKiZe** [reply]
dhcp succeeds, how would it know which subnet you want to load the file from?

**Reply by NiKiZe** [reply]
Reading your initial question again about selecting second NIC means that you want to use undionly.kpxe or snponly.efi to only get that one interface.

**Comment by wf2020** [comment]
Well, the file should be loaded from subnet 192.168.2.x but an IP address at subnet 192.168.3.x (netmask 255.255.255.0) is provided. So it would make sense to try the other available net.

**Comment by NiKiZe** [comment, marked as answer]
When ipxe.pxe or ipxe.efi is used all NICs that iPXE has drivers for is made available. Consideration is not taken for which NIC iPXE was loaded over.

If you want this to be taken into consideration then use undionly.kpxe or snponly.efi which should only make that one NIC available inside iPXE.

For any other magic about detecting networks, you will have to implement that logic yourself. Looping over interfaces, comparing subnet, retrying etc.

**Reply by wf2020** [reply]
If snponly.efi only uses the NIC iPXE was loaded over that perfectly makes sense for me.

**Reply by wf2020** [reply]
Thank you for pointing me to snponly.efi.
Now it works as I wish.
